lib/httpClient.js

```diff
@@ -189,6 +189,10 @@ HttpClient.prototype._requestHelper = function (options, done) {
       limits = rateLimiting.updateRateLimits(response.headers)
       return Promise.resolve(response.body)
     })
+    .catch((e) => {
+      limits = rateLimiting.updateRateLimits(e.response.headers)
```

Comment:
Thanks for the PR!

Since failures could happen for potentially all kinds of reasons, I think it's best to not to assume that that the error returned has a `response` property. Safer logic would be to either only call this method if e.response and e.response.headers exist, or wrap the method call in an additional `try` block and ignore the error if it fails.

Reply:
Ou are right. I updated the PR.

You will need to "Squash and merge" directly from Github